lib/ClangImporter/ClangDerivedConformances.cpp

```diff
@@ -435,55 +429,64 @@ void swift::conformToCxxIteratorIfNeeded(
   if (!iteratorCategory)
     return;
 
+  // In C++20, `std::contiguous_iterator_tag` is specified as a type called
+  // `iterator_concept`. It is not possible to detect a contiguous iterator
+  // based on its `iterator_category`. The type might not have an
+  // `iterator_concept` defined.
+  auto iteratorConcept = getIteratorConceptDecl(clangDecl);
+
+  auto unwrapUnderlyingTypeDecl =
+      [](clang::TypeDecl *typeDecl) -> clang::CXXRecordDecl * {
+    clang::CXXRecordDecl *underlyingDecl = nullptr;
+    if (auto typedefDecl = dyn_cast<clang::TypedefNameDecl>(typeDecl)) {
```

Comment:
Could we have multiple layers of `TypedefNameDecl`s? Would that be a problem?
E.g.:
```
struct MyIterator {
  using iterator_concept =  std::vector<int>::iterator::iterator_concept;
// ...
private:
  std::vector<int>::iterator it;
};
```

Reply:
Good catch! Let's make sure this works. I'll add a test.

Reply:
Actually this already works fine, I'll add a test in a separate PR.